tests/test_scan_pe.py

```diff
@@ -0,0 +1,194 @@
+
+    @unittest.skip("Need a PE file to test")
+    def test_resources_none(self):
+        self.assertIn("resources", self.scanner.metadata, "Resources were not scanned")
+        self.assertListEqual(self.scanner.metadata["resources"], [], "Wrong number of resources extracted")
+        self.assertIn(f"{self.scanner.scanner_name}::no_resources", self.file_object.flags, "Flag not properly applied")
+
+    def test_sections(self):
+        self.assertIn("sections", self.scanner.metadata, "Sections were not scanned")
+        sections = self.scanner.metadata["sections"]
+        self.assertEqual(self.scanner.metadata["total"]["sections"], 10, "Wrong number of sections extracted")
+        self.assertEqual(self.scanner.metadata["total"]["sections"], len(sections), "Section list length and total counted sections do not match")
+        section_names = [i["name"] for i in sections]
+        expected = [".00cfg", ".rdata", ".bss", ".data", ".gfids", ".rsrc", ".text", ".xdata", ".idata", ".reloc"]
+        self.assertListEqual(sorted(expected), sorted(section_names), "Wrong section name")
+        for section in sections:
+            self.assertEqual("IMAGE_SECTION_HEADER", section["structure"], "Section structure should be of type 'IMAGE_SECTION_HEADER'")
+
+    def test_signature(self):
+        child_names = [c.filename for c in self.scanner.children]
+        self.assertIn(f"{self.scanner.scanner_name}::digital_signature", child_names, "Digital signature not extracted")
+        self.assertIn(f"{self.scanner.scanner_name}::signed", self.file_object.flags, "Flag not properly applied")
+
+    @unittest.skip("Need a PE file to test")
+    def test_signature_empty(self):
+        self.assertIn(f"{self.scanner.scanner_name}::empty_signature", self.file_object.flags, "Flag not properly applied")
+
+    @unittest.skip("Need a PE file to test")
+    def test_signature_none(self):
+        self.assertNotIn(f"{self.scanner.scanner_name}::empty_signature", self.file_object.flags, "Flag not properly applied")
+        self.assertNotIn(f"{self.scanner.scanner_name}::signed", self.file_object.flags, "Flag not properly applied")
+
+    def test_version_info(self):
+        self.assertIn("versionInfo", self.scanner.metadata, "Version info was not scanned")
+        # only these entries should exist
+        # the test will fail if more or less entries are extracted
+        expected = {
+            "CompanyName": "Simon Tatham",
+            "ProductName": "PuTTY suite",
+            "FileDescription": "SSH, Telnet and Rlogin client",
+            "InternalName": "PuTTY",
+            "OriginalFilename": "PuTTY",
+            "FileVersion": "Release 0.70",
+            "ProductVersion": "Release 0.70",
+            "LegalCopyright": "Copyright \u00a9 1997-2017 Simon Tatham."
+        }
+        version_info = {}
+        for entry in self.scanner.metadata["versionInfo"]:
+            version_info[entry["string-name"]] = entry["value"]
```

Comment:
Finally got around to testing this for myself and I think the design is good for testing scanners (where the highest priority is ensuring the imported packages haven't altered outputted data). The only out of sync section I noticed when I ran the test is that the `string-name` field for the `versionInfo` dictionary should be `name` -- which is good, cause this means the unit test is doing its job! Feel free to make any additional changes you think are necessary and then I can merge when you're ready.

Reply:
Good catch, I've fixed the test, so go ahead and merge whenever